docs/arch.md

```diff
@@ -0,0 +1,286 @@
+And even if it is time to add to queue it balances inserts with AddAfter between i/4 to i/4 + random(i/4 * 2) (sometimes between 1/4 and 3/4).
+
+The consumers from the queue are 4 workers started as goroutines from periodic.Run function.
+Because we only have one gatherer in queue, I think only one worker will effectively run task from queue. The actual start of gatherer is call to c.sync from periodic.processNextWorkItem. The c.sync is also defined in periodic and calls Gather + immediatelly stores returned data to disk as one file with timestamp.
+
+### Scheduling and running of Uploader
+The operator.go is starting background task defined in pkg/insights/insightsuploader/insightsuploader.go. The insights uploader is periodically checking if there are any data to upload by calling summarizer. 
+if any exists. If no data to upload are found the uploader continues with next cycle.
+The uploader cycle is running wait.Poll function which is waiting until config changes or until there is a time to upload. The time to upload is determined set to initialDelay.
+If this is the first upload (the lastReportedTime from status is not set) the uploader uses interval/8+random(interval/8*2) as next upload time. This could be reset though to 0, if it is Safe to upload immediatelly. If any upload was already reported, the next upload interval is going to be now - lastReported + interval + 1.2 Jitter.
+Code: This line sets next interval in regular polling:
+```
+wait.Jitter(now.Sub(next), 1.2)
+```
+After calculation of initialDelay, wait.Until runs regular function, which starts waiting on either config change or until time till initialDelay and then continues. Every event is retriggered by wait.Until again every 15 seconds. For example if ClusterVersion is not populated (because Gatherer havent finished initial Gathering), it will retry in 15 seconds.
+Eventually uploader will use insightsclient.Send to run the upload itself. It then reports any errors to its Status reporter.
+
+# How is Uploader authenticating to cloud.redhat.com
+The insightsclient.Send is creating http client with Get method and url, which can be configured in config/pod.yaml, or eventually from support secret endpoint value. The transport is encrypted with TLS, which is set in clientTransport() method. This method is using pkg/authorizer/clusterauthorizer.go to
+add the Bearer token, which is read from secret pull-secret, the section .auths.cloud.redhat.com.auth. The clientTransport is also setting Proxy, which
+can be either from Proxy settings or from support secret, or from Env variables (cluster-wide).
+
+## Summarising the content before upload
+Summarizer is defined by pkg/record/diskrecorder/diskrecorder.go and is merging all existing archives. That means all archives with insights-*.tar.gz pattern which weren't removed and since last check are included. Then mergeReader is taking one file after another and adding all of them to archive under their path.
```

Comment:
> That means all archives with insights-*.tar.gz pattern which weren't removed and since last check are included. 

We should try to update this sentence somehow IMO :)